tests/conftest.py

```diff
@@ -260,40 +273,116 @@ def decam_example_raw_image( decam_example_exposure ):
+    test files were created, the files may not be exactly consistent
+    with the code_version in the provenances they have attached to them,
+    but hopefully the actual data and database records are still good
+    enough for the tests.
+
+    Has an image (with weight and flags), sources, and psf.  The data
+    has not been loaded into the image, sources, or psf fields, but of
+    course that will happen if you access (for instance)
+    decam_example_reduced_image_ds.image.data.  The provenances *have*
+    been loaded into the session (and committed to the database), but
+    the image, sources, and psf have not been added to the session.
+
+    The DataStore has a session inside it.
+
+    """
+
+    exposure = decam_example_exposure
     # Have to spoof the md5sum field to let us add it to the database even
     # though we're not really saving it to the archive
     exposure.md5sum = uuid.uuid4()
-    with SmartSession() as session:
-        # Spoof md5sum so we can add it
-        # Think: may be better just to actually save the exposure
-        # to the test archive?
-        exposure = exposure.recursive_merge( session )
-        session.add( exposure )
-        session.commit()
-        prepper = Preprocessor()
-        # NOTE: this has a side effect of loading some DECam
-        # calibration files (flats, etc.) into the databse
-        ds = prepper.run( exposure, 'N1', session=session )
-        ds.save_and_commit( session=session )
-    yield ds
-    ds.delete_everything()
 
-@pytest.fixture(scope="session")
-def decam_example_reduced_image_source_list_ds( decam_example_reduced_image_ds ):
-    """Returns the same datastore from decam_example_reduced_image_ds, only now with a source list too"""
-    det = Detector()
-    ds = det.run( decam_example_reduced_image_ds )
-    ds.save_and_commit()
-    yield ds
-    # This next line may not be necessary since
-    # decam_example_reduced_image_ds will run it
-    # ...indeed, it looks like data_store.delete_everyting()
-    # is not robust to being called more than once
-    # ds.delete_everything()
+    # This next block of code generates the files read in this test
+    #   (though they'll be written to data/115 instead of
+    #   data/test_data/DECam_examples).  Commented out here because
+    #   of course we just want to read them, but leave the commented
+    #   code here in case for whatever reason we want to regenerate them.
+    #   (In that case, run up to the pdb interrupt, and then copy the
+    #   files you want out of data/115.)
+    # with SmartSession() as session:
+    #     # Spoof md5sum so we can add it
+    #     exposure = exposure.recursive_merge( session )
+    #     session.add( exposure )
+    #     session.commit()
+    #     prepper = Preprocessor()
+    #     ds = prepper.run( exposure, 'N1', session=session )
+    #     det =  Detector()
+    #     ds = det.run( ds )
+    #     ds.save_and_commit()
+    # import pdb; pdb.set_trace()
```

Comment:
Yeah, I'll do it so that it looks for the _cached files, and regenerates them if it's not there.  Then I'll git rm the _cached files; I really hope that when we merge with rebase to main, it doesn't stick those giant files in the commit history.  (I believe that that's part of the point of rebase.)

Reply:
(...or, rather, I mean, squash.)

Reply:
Yeah I think the squash will kill the files.